tests/units/test_diff_non_default_encoding.rb

```diff
@@ -35,13 +35,15 @@ def setup
   def test_diff_with_greek_encoding
     d = @git.diff
     patch = d.patch
+    return unless Encoding.default_external == (Encoding::UTF_8 rescue Encoding::UTF8) # skip test on Windows / check UTF8 in JRuby instead
```

Comment:
I don't think this is quite right.  What do you think about this:

```Ruby
skip 'Encoding tests do not work in Windows' if windows_platform?
```

On other platforms, these tests should work not matter what the external encoding is.

Reply:
I forgot to mention... I have not ever seen `rescue` used in this way. Can you point me to something that explains how you are using it?

Reply:
About your first question, if you can figure out a better way to skip tests on Windows, then power to you! After all, they do not work in their current form inside the Windows Command Prompt or Git Bash and I did not feel it was important enough for me to invest more time into it (especially given that there was zero coverage before to begin with, so any coverage now is better than nothing)

About your second question, I learned this from a senior Ruby developer called [Chris Sepic](https://github.com/cgs) more than a decade ago while working at this Chicago suburb company, Leapfrog Online.

Basically, in rare cases where you do not need to do anything with the error exception object, instead of doing this:

```ruby
begin 
  try_something
rescue => e
  do_nothing_with_e
end
```

You can drop the error exception object:

```ruby
begin 
  try_something
rescue
  do_something
end
```

And then finally, you can fold it into one line:

```ruby
try_something rescue do_something
```

I don't advise using this in the majority of cases. I only used it because this was a test (not production code) and brevity helped keep things on one line.

Specifically, there is this odd quirk about JRuby where it calls the encoding constant UTF8 instead of UTF_8, so instead of me writing something very long and elaborate to check it such as:

```
if Encoding.constants.include?(:UTF_8)
  # do something with Encoding::UTF_8
elsif Encoding.constants.include?(:UTF8)
  # do something with Encoding::UTF8
end

I just shortened it knowing that if you call `Encoding::UTF_8` in JRuby, it bombs, and then the `rescue` keyword rescues it unto the next value to compare to, which is `Encoding::UTF8`

Again, this is not a recommended technique in general. I just did it because it was low-risk code in a test (spec).